File: binance4j-core/src/main/java/com/binance4j/core/annotation/PostRequest.java
package com.binance4j.core.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import com.binance4j.core.Request;
import com.binance4j.core.dto.HttpMethod;
import com.binance4j.core.dto.RateLimitType;
import com.binance4j.core.dto.Signature;

/** {@link Request} metadata. */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@ApiRequest(method = HttpMethod.POST)
public @interface PostRequest {

	/** @return The request signature. */
	Signature signature() default Signature.NONE;

	/** @return Is the request an order? */
	boolean isOrder() default false;

	/** @return The request weight. */
	int weight() default 1;

	/** @return The request rate limit. */
	RateLimitType rateLimit() default RateLimitType.IP;

	/** @return The request path. */
	String path() default "";
}

File: connectors/src/main/kotlin/com/binance4j/core/annotation/PostRequest.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.core.annotation

import com.binance4j.core.dto.HttpMethod
import com.binance4j.core.dto.RateLimitType
import com.binance4j.core.dto.Signature

/** [Request] metadata.  */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@ApiRequest(method = HttpMethod.POST)
annotation class PostRequest(
    /** The request signature. */
    val signature: Signature = Signature.NONE,
    /** Is the request an order? */
    val isOrder: Boolean = false,
    /** The request weight. */
    val weight: Int = 1,
    /** The request rate limit. */
    val rateLimit: RateLimitType = RateLimitType.IP,
    /** The request path. */
    val path: String = ""
)